src/pudl/helpers.py

```diff
@@ -916,24 +919,36 @@ def convert_cols_dtypes(df, data_source, name=None):
         :mod:`pudl.constants` ``column_dtypes`` dictionary.
 
     """
+    source_dtypes = pc.column_dtypes[data_source]
```

Comment:
Is this just for readability? Or does it have a logical consequence?

Reply:
for readability/with the addition of the check for non-nullable strings we were pulling this 3 or 4 times.